### ISSUE TITLE
Keep getting this error: Cannot read property 'kind' of undefined

### ISSUE THREAD

**Issue by therawk** [angular/angular-cli#13172]
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [x ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

Angular CLI: 7.1.2
Node: 8.11.4
OS: win32 x64
Angular: 7.1.2
... animations, cli, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.11.2
@angular-devkit/build-angular     0.11.2
@angular-devkit/build-optimizer   0.11.2
@angular-devkit/build-webpack     0.11.2
@angular-devkit/core              7.1.2
@angular-devkit/schematics        7.1.2
@angular/cdk                      7.1.1
@ngtools/webpack                  7.1.2
@schematics/angular               7.1.2
@schematics/update                0.11.2
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.23.1

### Repro steps
Happens many times a day when saving a file and the cli re-compiles

### The log given by the failure
Failed to compile.

: TypeError: Cannot read property 'kind' of undefined
    at getAssignmentTargetKind (node_modules\typescript\lib\typescript.js:10049:28)
    at Object.isAssignmentTarget (node_modules\typescript\lib\typescript.js:10092:16)
    at checkObjectLiteral (node_modules\typescript\lib\typescript.js:46107:45)
    at checkExpressionWorker (node_modules\typescript\lib\typescript.js:50748:28)
    at checkExpression (node_modules\typescript\lib\typescript.js:50696:42)
    at checkExpressionWithContextualType (node_modules\typescript\lib\typescript.js:50524:26)
    at checkApplicableSignature (node_modules\typescript\lib\typescript.js:47963:35)
    at chooseOverload (node_modules\typescript\lib\typescript.js:48281:26)
    at resolveCall (node_modules\typescript\lib\typescript.js:48231:26)
    at resolveCallExpression (node_modules\typescript\lib\typescript.js:48534:20)
    at resolveSignature (node_modules\typescript\lib\typescript.js:48800:28)
    at getResolvedSignature (node_modules\typescript\lib\typescript.js:48844:26)
    at checkCallExpression (node_modules\typescript\lib\typescript.js:48936:29)
    at checkExpressionWorker (node_modules\typescript\lib\typescript.js:50759:28)
    at checkExpression (node_modules\typescript\lib\typescript.js:50696:42)
    at resolveDecorator (node_modules\typescript\lib\typescript.js:48741:28)

### Desired functionality
No errors

Details:
This started when Angular 7 was released. Unfortunately I wasn't able to find much information about this issue. Neither on Stackoverflow nor in the issue trackers. Well at least not since the release of Angular 7

**Comment by djleonskennedy** [comment]
@therawk have same @angular-devkit/build-angular 0.11.2, full recompile fixes it

**Comment by alan-agius4** [comment]
We’d need a bit more to look into this for example a code snippet that is causing this error.

If you instrument the `checkSourceFile` function in `typescript.js` to print `node.path` on entry, you can at least check what file started the chain of operations that led to the crash. Typescript don’t have any builtin functionality for tracking checker crashes.

**Comment by therawk** [comment]
@alan-agius4 Uh :| 
To be honest I have no idea how to do that. I searched google but the only thing I found are actual error logs. 
Would you mind telling me what I need to do? Sorry

**Comment by alan-agius4** [comment]
@therawk 

1. Open `node_modules/typescript/lib/typescript.js`
2. locate `function checkSourceFile` 
3. Do a `console.log(node.path)`

Execute build/serve command.

**Comment by therawk** [comment]
I just tried this and here's the output. Does this help at all?

NodeObject {
  pos: 1683,
  end: 1834,
  flags: 8,
  transformFlags: 536870912,
  parent: undefined,
  kind: 186,
  properties:
   [ NodeObject {
       pos: 1684,
       end: 1721,
       flags: 0,
       transformFlags: 536870912,
       parent: [Object],
       kind: 273,
       decorators: undefined,
       modifiers: undefined,
       name: [Object],
       questionToken: undefined,
       exclamationToken: undefined,
       initializer: [Object],
       modifierFlagsCache: 536870912,
       symbol: [Object] },
     NodeObject {
       pos: 1722,
       end: 1776,
       flags: 8,
       transformFlags: 536870912,
       parent: undefined,
       kind: 273,
       name: [Object],
       questionToken: undefined,
       initializer: [Object],
       original: [Object],
       modifierFlagsCache: 536870912 },
     NodeObject {
       pos: -1,
       end: -1,
       flags: 8,
       transformFlags: 536870912,
       parent: undefined,
       kind: 273,
       name: [Object],
       questionToken: undefined,
       initializer: [Object],
       modifierFlagsCache: 536870912 },
     pos: -1,
     end: -1,
     hasTrailingComma: undefined,
     transformFlags: 536870912 ],
  multiLine: true,
  original:
   NodeObject {
     pos: 1683,
     end: 1834,
     flags: 0,
     transformFlags: 536870912,
     parent:
      NodeObject {
        pos: 1673,
        end: 1835,
        flags: 8192,
        transformFlags: 536870912,
        parent: [Object],
        kind: 189,
        expression: [Object],
        arguments: [Array],
        id: 116507 },
     kind: 186,
     multiLine: true,
     properties:
      [ [Object],
        [Object],
        [Object],
        pos: 1684,
        end: 1831,
        transformFlags: 536870912 ],
     symbol:
      SymbolObject {
        flags: 4096,
        escapedName: '__object',
        declarations: [Array],
        members: [Object],
        valueDeclaration: [Circular] },
     nextContainer:
      NodeObject {
        pos: 3444,
        end: 4790,
        flags: 128,
        transformFlags: 537182338,
        parent: [Object],
        kind: 155,
        decorators: undefined,
        modifiers: undefined,
        typeParameters: undefined,
        parameters: [Array],
        body: [Object],
        modifierFlagsCache: 536870912,
        symbol: [Object],
        locals: [Object],
        nextContainer: [Object],
        returnFlowNode: [Object],
        id: 116509 },
     contextualType: undefined,
     contextualMapper: undefined,
     jsDocCache: [] },
  modifierFlagsCache: 536870912,
  contextualType:
   TypeObject {
     checker:
      { getNodeCount: [Function: getNodeCount],
        getIdentifierCount: [Function: getIdentifierCount],
        getSymbolCount: [Function: getSymbolCount],
        getTypeCount: [Function: getTypeCount],
        isUndefinedSymbol: [Function: isUndefinedSymbol],
        isArgumentsSymbol: [Function: isArgumentsSymbol],
        isUnknownSymbol: [Function: isUnknownSymbol],
        getMergedSymbol: [Function: getMergedSymbol],
        getDiagnostics: [Function: getDiagnostics],
        getGlobalDiagnostics: [Function: getGlobalDiagnostics],
        getTypeOfSymbolAtLocation: [Function: getTypeOfSymbolAtLocation],
        getSymbolsOfParameterPropertyDeclaration: [Function: getSymbolsOfParameterPropertyDeclaration],
        getDeclaredTypeOfSymbol: [Function: getDeclaredTypeOfSymbol],
        getPropertiesOfType: [Function: getPropertiesOfType],
        getPropertyOfType: [Function: getPropertyOfType],
        getTypeOfPropertyOfType: [Function: getTypeOfPropertyOfType],
        getIndexInfoOfType: [Function: getIndexInfoOfType],
        getSignaturesOfType: [Function: getSignaturesOfType],
        getIndexTypeOfType: [Function: getIndexTypeOfType],
        getBaseTypes: [Function: getBaseTypes],
        getBaseTypeOfLiteralType: [Function: getBaseTypeOfLiteralType],
        getWidenedType: [Function: getWidenedType],
        getTypeFromTypeNode: [Function: getTypeFromTypeNode],
        getParameterType: [Function: getTypeAtPosition],
        getPromisedTypeOfPromise: [Function: getPromisedTypeOfPromise],
        getReturnTypeOfSignature: [Function: getReturnTypeOfSignature],
        getNullableType: [Function: getNullableType],
        getNonNullableType: [Function: getNonNullableType],
        typeToTypeNode: [Function: typeToTypeNode],
        indexInfoToIndexSignatureDeclaration: [Function: indexInfoToIndexSignatureDeclaration],
        signatureToSignatureDeclaration: [Function: signatureToSignatureDeclaration],
        symbolToEntityName: [Function: symbolToEntityName],
        symbolToExpression: [Function: symbolToExpression],
        symbolToTypeParameterDeclarations: [Function: symbolToTypeParameterDeclarations],
        symbolToParameterDeclaration: [Function: symbolToParameterDeclaration],
        typeParameterToDeclaration: [Function: typeParameterToDeclaration],
        getSymbolsInScope: [Function: getSymbolsInScope],
        getSymbolAtLocation: [Function: getSymbolAtLocation],
        getShorthandAssignmentValueSymbol: [Function: getShorthandAssignmentValueSymbol],
        getExportSpecifierLocalTargetSymbol: [Function: getExportSpecifierLocalTargetSymbol],
        getExportSymbolOfSymbol: [Function: getExportSymbolOfSymbol],
        getTypeAtLocation: [Function: getTypeAtLocation],
        getPropertySymbolOfDestructuringAssignment: [Function: getPropertySymbolOfDestructuringAssignment],
        signatureToString: [Function: signatureToString],
        typeToString: [Function: typeToString],
        symbolToString: [Function: symbolToString],
        typePredicateToString: [Function: typePredicateToString],
        writeSignature: [Function: writeSignature],
        writeType: [Function: writeType],
        writeSymbol: [Function: writeSymbol],
        writeTypePredicate: [Function: writeTypePredicate],
        getAugmentedPropertiesOfType: [Function: getAugmentedPropertiesOfType],
        getRootSymbols: [Function: getRootSymbols],
        getContextualType: [Function: getContextualType],
        getContextualTypeForObjectLiteralElement: [Function: getContextualTypeForObjectLiteralElement],
        getContextualTypeForArgumentAtIndex: [Function: getContextualTypeForArgumentAtIndex],
        getContextualTypeForJsxAttribute: [Function: getContextualTypeForJsxAttribute],
        isContextSensitive: [Function: isContextSensitive],
        getFullyQualifiedName: [Function: getFullyQualifiedName],
        getResolvedSignature: [Function: getResolvedSignature],
        getResolvedSignatureForSignatureHelp: [Function: getResolvedSignatureForSignatureHelp],
        getConstantValue: [Function: getConstantValue],
        isValidPropertyAccess: [Function: isValidPropertyAccess],
        isValidPropertyAccessForCompletions: [Function: isValidPropertyAccessForCompletions],
        getSignatureFromDeclaration: [Function: getSignatureFromDeclaration],
        isImplementationOfOverload: [Function: isImplementationOfOverload],
        getImmediateAliasedSymbol: [Function: getImmediateAliasedSymbol],
        getAliasedSymbol: [Function: resolveAlias],
        getEmitResolver: [Function: getEmitResolver],
        getExportsOfModule: [Function: getExportsOfModuleAsArray],
        getExportsAndPropertiesOfModule: [Function: getExportsAndPropertiesOfModule],
        getSymbolWalker: [Function: getSymbolWalker],
        getAmbientModules: [Function: getAmbientModules],
        getAllAttributesTypeFromJsxOpeningLikeElement: [Function: getAllAttributesTypeFromJsxOpeningLikeElement],
        getJsxIntrinsicTagNamesAt: [Function: getJsxIntrinsicTagNamesAt],
        isOptionalParameter: [Function: isOptionalParameter],
        tryGetMemberInModuleExports: [Function: tryGetMemberInModuleExports],
        tryGetMemberInModuleExportsAndProperties: [Function: tryGetMemberInModuleExportsAndProperties],
        tryFindAmbientModuleWithoutAugmentations: [Function: tryFindAmbientModuleWithoutAugmentations],
        getApparentType: [Function: getApparentType],
        getUnionType: [Function: getUnionType],
        createAnonymousType: [Function: createAnonymousType],
        createSignature: [Function: createSignature],
        createSymbol: [Function: createSymbol],
        createIndexInfo: [Function: createIndexInfo],
        getAnyType: [Function: getAnyType],
        getStringType: [Function: getStringType],
        getNumberType: [Function: getNumberType],
        createPromiseType: [Function: createPromiseType],
        createArrayType: [Function: createArrayType],
        getBooleanType: [Function: getBooleanType],
        getFalseType: [Function: getFalseType],
        getTrueType: [Function: getTrueType],
        getVoidType: [Function: getVoidType],
        getUndefinedType: [Function: getUndefinedType],
        getNullType: [Function: getNullType],
        getESSymbolType: [Function: getESSymbolType],
        getNeverType: [Function: getNeverType],
        isSymbolAccessible: [Function: isSymbolAccessible],
        isArrayLikeType: [Function: isArrayLikeType],
        isTypeInvalidDueToUnionDiscriminant: [Function: isTypeInvalidDueToUnionDiscriminant],
        getAllPossiblePropertiesOfTypes: [Function: getAllPossiblePropertiesOfTypes],
        getSuggestionForNonexistentProperty: [Function: getSuggestionForNonexistentProperty],
        getSuggestionForNonexistentSymbol: [Function: getSuggestionForNonexistentSymbol],
        getSuggestionForNonexistentExport: [Function: getSuggestionForNonexistentExport],
        getBaseConstraintOfType: [Function: getBaseConstraintOfType],
        getDefaultFromTypeParameter: [Function: getDefaultFromTypeParameter],
        resolveName: [Function: resolveName],
        getJsxNamespace: [Function: getJsxNamespace],
        getAccessibleSymbolChain: [Function: getAccessibleSymbolChain],
        getTypePredicateOfSignature: [Function: getTypePredicateOfSignature],
        resolveExternalModuleSymbol: [Function: resolveExternalModuleSymbol],
        tryGetThisTypeAt: [Function: tryGetThisTypeAt],
        getTypeArgumentConstraint: [Function: getTypeArgumentConstraint],
        getSuggestionDiagnostics: [Function: getSuggestionDiagnostics],
        runWithCancellationToken: [Function: runWithCancellationToken],
        getLocalTypeParametersOfClassOrInterfaceOrTypeAlias: [Function: getLocalTypeParametersOfClassOrInterfaceOrTypeAlias] },
     flags: 131072,
     id: 13634,
     objectFlags: 2,
     symbol:
      SymbolObject {
        flags: 66,
        escapedName: 'Component',
        declarations: [Array],
        members: [Object],
        parent: [Object],
        exports: Map {},
        valueDeclaration: [Object],
        id: 100970,
        isReferenced: 67897832 },
     members:
      Map {
        'changeDetection' => [Object],
        'viewProviders' => [Object],
        'moduleId' => [Object],
        'templateUrl' => [Object],
        'template' => [Object],
        'styleUrls' => [Object],
        'styles' => [Object],
        'animations' => [Object],
        'encapsulation' => [Object],
        'interpolation' => [Object],
        'entryComponents' => [Object],
        'preserveWhitespaces' => [Object],
        'selector' => [Object],
        'inputs' => [Object],
        'outputs' => [Object],
        'providers' => [Object],
        'exportAs' => [Object],
        'queries' => [Object],
        'jit' => [Object],
        'host' => [Object] },
     properties:
      [ [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object] ],
     callSignatures: [],
     constructSignatures: [],
     stringIndexInfo: undefined,
     numberIndexInfo: undefined,
     resolvedBaseTypes: [ [Object] ],
     declaredProperties:
      [ [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object] ],
     declaredCallSignatures: [],
     declaredConstructSignatures: [],
     declaredStringIndexInfo: undefined,
     declaredNumberIndexInfo: undefined },
  contextualMapper: undefined }

**Comment by alan-agius4** [comment]
It looks like it's happening for a `Component`, I tried to do a couple of test to replicate this, however so far I didn't manage to came across with this error.

**Comment by alan-agius4** [comment]
By any chance do you have `forkTypeChecker` set to false in your angular.json?

**Comment by therawk** [comment]
Ok. I'll migrate my project to a new one - wanted to re-work the design anyway. If I notice something I'll report here. 

The error is just annoying. As @djleonskennedy said recompiling fixes it

**Comment by mgechev** [comment]
Let us know (in a new issue) if you hit this issue again and you notice a pattern for reproduction.

**Comment by harshsyno** [comment]
I have encountered same issue and it got resolved when I executed command ng update. It happens as upgrading angular version also requires update angular/cli.

**Comment by therawk** [comment]
@harshsyno 
The problem is, this still happens when I tried adding my old components/modules to a completely new project using the latest angular-cli :|

Still having this issue. It's so annoying

**Comment by harshsyno** [comment]
@therawk 
It is happening mainly because with Latest Angular-cli and Angular 7, folder structure might be changed as the way of importing external files are only supported with Assets directory of Project Structure.

**Comment by rkachhawaha** [comment]
Simply stop application and execute "npm install" again. It fixed my issue.

**Comment by palpatine1991** [comment]
The problem in our case was that the locally installed version of @angular/cli was different to globally installed version of @angular/cli.

**Comment by evilgeniuscreative** [comment]
I am getting this issue as well despite updates and making sure that local/global cli are same version

**Comment by qaxqaxqax** [comment]
I am still getting this issue with ngx-build-plus.
I simply use the _ng build --watch --project=dummy-project_ and the recompile turns into this error.

@angular-devkit/architect  0.13.6
@angular-devkit/build-angular  0.13.6
@angular-devkit/build-optimizer  0.13.6
@angular-devkit/build-webpack  0.13.6
@angular-devkit/core  7.3.6
@angular-devkit/schematics  7.3.6
@angular/cli  7.3.6
@ngtools/webpack  7.3.6
@schematics/angular 7.3.6
@schematics/update  0.13.6
rxjs  6.4.0
typescript  3.2.4
webpack 4.29.6

**Comment by ehsanagh** [comment]
Hi All, I just installed angular ssr and this is happening to me too.

**Comment by palpatine1991** [comment]
Ignore my previous comment. It was just high hope. It did not happen for a while so I thought it is fixed but the error started to appear again.

**Comment by khoa3248** [comment]
Your source contains var x , when edit to var x= int or string or any then error fixed

**Comment by myWorkingGithub** [comment]
The same mistake in Angular 7 project. Observation has shown for several weeks that this may occur at different moments of work, the strangest of them, when the application is just running locally, I do not work in it for several hours, after a while, the application automatically recompiles and I get this error

**Comment by izzeda** [comment]
@myWorkingGithub: I have the same thing: usually when the app is running locally, and I haven't touched it for about half an hour, I get the error... Also Angular 7.

**Comment by shripadgodse** [comment]
did anyone has found solution for that problem? if yes please let me know i am facing same problem

**Comment by dsldiesel** [comment]
Please solution!
This error appears randomly when updating a ts file that imports swal!!!

**Comment by harshsyno** [comment]
@myWorkingGithub, @izzeda, @shripadgodse, @dsldiesel 
Kindly Stop compiler with cntr+c & Recompile entire project.
This will end the error.

**Comment by shripadgodse** [comment]
> @myWorkingGithub, @izzeda, @shripadgodse, @dsldiesel
> Kindly Stop compiler with cntr+c & Recompile entire project.
> This will end the error.

But this is temporary solution. Same issue is happening after some time

**Comment by klinki** [comment]
I'm getting the same error. Also on Angular 7 (7.2.5) with TypeScript 3.2.2.

It happens randomly after changing a file and recompilation.

**Comment by harshsyno** [comment]
> > @myWorkingGithub, @izzeda, @shripadgodse, @dsldiesel
> > Kindly Stop compiler with cntr+c & Recompile entire project.
> > This will end the error.
> 
> But this is temporary solution. Same issue is happening after some time

If you replace any file while the server is initialized then you will get these error every time. If you just edit your file then it would simply recompile.
@klinki 
You should also stop the server before changing/replacing the file.

**Comment by klinki** [comment]
@harshsyno I see. I don't usually replace files. (I assume by replacing you mean deleting the file and creating it again?). But this might be related to git (maybe changing branch or something...). I'd expect it to just update the file, but maybe that's not the case.

**Comment by harshsyno** [comment]
> @harshsyno I see. I don't usually replace files. (I assume by replacing you mean deleting the file and creating it again?). But this might be related to git (maybe changing branch or something...). I'd expect it to just update the file, but maybe that's not the case.

@klinki I understand your concern. But the solution I got working is simply starting the server again.

**Comment by make-ing** [comment]
Same problem here... but it works when i only do a normal `ng serve`. But when i try to do a `ng serve --configuration=prodution` it fails with the same error message mentioned above.

**Comment by tasneem995** [comment]
just do ionic serve again 
it's worked for me

**Comment by jeandat** [comment]
What's the reason to close this issue. It seems to me this is not resolved at all.

**Comment by alan-agius4** [comment]
Hi all, the above issue should have been solved with https://github.com/angular/angular-cli/pull/13992 which is available in `@ngtools/webpack `7.3.8`. 

Please update to the most recent Angular CLI version.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.

Thanks.

**Comment by PierreChavaroche** [comment]
@jer-tx , I had the same problem. 

What helped to me was to delete both "node_modules" and "package-lock.json". Then I run the command 
`ng add @nguniversal/express-engine --clientProject <angular-project>`
and it complained about missing node modules. I installed them manually one by one (only what was really necessary) . And finally it worked!
After that you can re-install everything else by 
`npm run i`
